File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/java/subsystems/Robot.java
package org.firstinspires.ftc.teamcode.java.subsystems;

import java.util.Set;

import dev.nextftc.robot.Mechanism;
import dev.nextftc.robot.NextRobot;

public class Robot implements NextRobot {
    public final Claw claw = new Claw();
    public final Slides slides = new Slides();
    public final Drivetrain drivetrain = new Drivetrain();

    @Override
    public Set<Mechanism> getMechanisms() {
        return Set.of(claw, slides, drivetrain);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/kotlin/subsystems/Robot.kt
package org.firstinspires.ftc.teamcode.kotlin.subsystems

import dev.nextftc.robot.Mechanism
import dev.nextftc.robot.NextRobot

class Robot : NextRobot {
    val claw = Claw()
    val slides = Slides()
    val drivetrain = Drivetrain()

    override val mechanisms: Set<Mechanism>
        get() = setOf(claw, slides, drivetrain)
}